src/Features/CSharp/Portable/MoveToResx/CSharpMoveToResxCodeFixProvider.cs

```diff
@@ -0,0 +1,474 @@
+            if (!documentsAndDiagnosticsToFixMap.TryGetValue(document, out var diagnostics) || diagnostics.IsEmpty)
+                return null;
+
+            var project = document.Project;
+            var allResx = GetAllResxFiles(project).ToList();
+            if (allResx.Count == 0)
+                return null;
+
+            var resx = allResx.First();
+
+            return CodeAction.Create(
+                CSharpFeaturesResources.Move_all_strings_to_resx_resource,
+                async ct =>
+                {
+                    // Collect all string literals and their operations
+                    var root = await document.GetRequiredSyntaxRootAsync(ct).ConfigureAwait(false);
+                    var resourceOperations = new List<ResourceOperation>();
+                    var replacementOperations = new List<ReplacementOperation>();
+
+                    // First pass: collect all operations and determine actual resource keys
+                    var resxSourceText = await resx.GetTextAsync(ct).ConfigureAwait(false);
+                    XDocument xdoc;
+                    try
+                    {
+                        xdoc = XDocument.Parse(resxSourceText.ToString());
+                    }
+                    catch (System.Xml.XmlException)
+                    {
+                        return project.Solution;
+                    }
+
+                    if (xdoc.Root == null)
+                    {
+                        return project.Solution;
+                    }
+
+                    foreach (var diagnostic in diagnostics)
+                    {
+                        var node = root?.FindNode(diagnostic.Location.SourceSpan) as LiteralExpressionSyntax;
+                        if (node is not null)
+                        {
+                            var value = node.Token.ValueText;
+                            var proposedKey = ToDeterministicResourceKey(value);
+                            var resourceOperation = new ResourceOperation(value, proposedKey);
+
+                            // Get the actual key that will be used (may be existing key if value already exists)
+                            var actualKey = ResxResourceManager.GetActualResourceKey(xdoc, resourceOperation);
+
+                            resourceOperations.Add(new ResourceOperation(value, actualKey));
+                            replacementOperations.Add(new ReplacementOperation(node, actualKey));
+                        }
+                    }
+
+                    if (resourceOperations.Count == 0)
+                        return project.Solution;
+
+                    return await ResxResourceManager.UpdateResxAndDocumentAsync(
+                        document,
+                        resx,
+                        resourceOperations,
+                        replacementOperations,
+                        ct).ConfigureAwait(false);
+                },
+                nameof(CSharpMoveToResxFixAllProvider));
```

Comment:
all this code feels very duplicative with the fix-one case.  can you refactor to share logic?